provider/openstack/firewaller.go

```diff
@@ -46,14 +47,14 @@ type Firewaller interface {
 	// controller, ie those for all hosted models.
 	DeleteAllControllerGroups(controllerUUID string) error
 
+	// DeleteGroups deletes the security groups with the specified names.
+	DeleteGroups(names ...string) error
+
 	// Implementations should return list of security groups, that belong to given instances.
 	GetSecurityGroups(ids ...instance.Id) ([]string, error)
 
 	// Implementations should set up initial security groups, if any.
-	SetUpGroups(controllerUUID, machineId string, apiPort int) ([]neutron.SecurityGroupV2, error)
-
-	// Set of initial networks, that should be added by default to all new instances.
-	InitialNetworks() []nova.ServerNetworks
+	SetUpGroups(controllerUUID, machineId string, apiPort int) ([]nova.SecurityGroupName, error)
```

Comment:
If we're providing two implemntations of this, returning []string seems like the common point.